common/persistence/sql/storage/store.go

```diff
@@ -67,7 +79,83 @@ func NewSQLDB(cfg *config.SQL) (sqldb.Interface, error) {
 	}
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
-	return mysql.NewDB(db, nil), nil
+	return db, nil
+}
+
+func RegisterTLSConfig(cfg *config.SQL) error {
+	host, _, err := getHostPortFromConnectAddr(cfg.ConnectAddr)
+	if err != nil {
+		log.Panicf("error getting host port from ConnectAddr: %v", err)
+	}
+
+	tlsConfig := &tls.Config{
+		ServerName: *host, // hostname
+	}
+
+	if cfg.TLS.CaFile != "" {
+		rootCertPool := x509.NewCertPool()
+		pem, err := ioutil.ReadFile(cfg.TLS.CaFile)
+		if err != nil {
+			return fmt.Errorf("failed to load CA files: %v", err)
+		}
+		if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+			return fmt.Errorf("failed to append CA file")
+		}
+		tlsConfig.RootCAs = rootCertPool
+	}
+
+	if cfg.TLS.CertFile != "" && cfg.TLS.KeyFile != "" {
+		clientCert := make([]tls.Certificate, 0, 1)
+		certs, err := tls.LoadX509KeyPair(
+			cfg.TLS.CertFile,
+			cfg.TLS.KeyFile,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to load key pair: %v", err)
+		}
+		clientCert = append(clientCert, certs)
+		tlsConfig.Certificates = clientCert
+	}
+
+	// Register a customer tls configuration
+	err = mysql.RegisterTLSConfig(customTLSName, tlsConfig)
+	if err != nil {
+		return fmt.Errorf("failed to register tls config: %v", err)
+	}
+
+	if cfg.ConnectAttributes == nil {
+		cfg.ConnectAttributes = map[string]string{}
+	}
+
+	if cfg.ConnectAttributes["tls"] == customTLSName || cfg.ConnectAttributes["tls"] == "" {
+		// Use the new custom tls configuration
+		cfg.ConnectAttributes["tls"] = customTLSName
```

Comment:
I'm modifying the config here to use the new registered tls config. Not sure if this is best practice or not?

Reply:
sorry, I am missing where this modified connectAttributes is being used ?